python_tools/flipper_file_to_ones_and_zeroes.py

```diff
@@ -1,19 +1,101 @@
+                split_run_length_lists.append(run_length_list[start:i])
+            start = i + 1
+            skip = False
+        # TODO consider throwing out any codes with a too long string of ones
+        if val > max_ones * pulse_length and i % 2 == 0:
+            skip = True
+    if not skip:
+        split_run_length_lists.append(run_length_list[start:])
+    return split_run_length_lists
+
+
+def flipper_file_to_bits(filename):
+    run_length_lists = flipper_file_to_run_length_lists(filename)
+    split_run_length_lists = []
+    for run_length_list in run_length_lists:
+        split_run_length_lists += split_run_length_list(run_length_list)
+    bit_lists = []
+    for run_length_list in split_run_length_lists:
+        # A ValueError indicates that some value is not close enough to a multiple of pulse_length, so that code is
+        # thrown out.
+        try:
+            new_bit_list = funcs.run_lengths_to_bits(run_length_list, acceptable_error=.15,
+                                                     pulse_length=cfg.PULSE_LENGTH)
+            add_to_bit_lists_avoid_duplicates(bit_lists, new_bit_list)
+            # print(split_run_length_list)
+        except ValueError as e:
+            # print(e)
+            pass
+    return bit_lists
+
+
+def add_to_bit_lists_avoid_duplicates(bit_lists, new_bit_list):
+    # Check if the previous code is actually just the end of this code.
+    # This happens in the case where a recording started in the middle of an IR code.
+    if len(bit_lists) > 0:
+        previous = bit_lists[-1]
+        if new_bit_list[-1 * len(previous):] == previous:
+            bit_lists.remove(previous)
+    # Check for whole repeated codes
+    # Note: You'll have to check again before adding to the big main list, this just checks within the one file.
+    if new_bit_list not in bit_lists:
+        bit_lists.append(new_bit_list)
+        # print(new_bit_list)
+    else:
+        ...
+        # TODO increment some counter for each code
+
 
+def send_all_found_flipper_codes():
+    arduino = serial.Serial(port=cfg.ARDUINO_SERIAL_PORT, baudrate=cfg.ARDUINO_BAUD_RATE, timeout=.1)
+    mega_list = []
+    root_dir = "../raw_wild_ir_captures"
+    p = Path(root_dir)
+    for path in p.rglob("**/*.ir"):
+        codes = flipper_file_to_bits(str(path))
+        for code in codes:
+            if code not in mega_list:
+                mega_list.append(code)
+    for code in mega_list:
+        try:
+            print(code)
+            arduino_string_ver = funcs.bits_to_arduino_string(code)
+            arduino.write(bytes(arduino_string_ver, 'utf-8'))
+            input("Press enter for next")
+            print("Command sent to Arduino.")
+        except ValueError as e:
+            print(e)
 
-for l in flipper_file_to_binary("../raw_wild_ir_captures/cleveland_cavaliers_2022_home_opener_recorded_by_dani/Remote20.ir"):
-    print(l)
\ No newline at end of file
+if __name__ == "__main__":
```

Comment:
I like this pattern of having a way to run the file to do a thing but also just be able to import the functions separately :)